File: dotnet/eshop-cqrs-ddd-microservices-excercise/Com.Marekturis.Product2/Model/Infrastructure/Remote/RemoteAuthorizator.cs
using System.Net;
using Com.Marekturis.Product2.Model.Application.Authorization;
using RestSharp;

namespace Com.Marekturis.Product2.Infrastructure.Remote
{
    public class RemoteAuthorizator : Authorizator
    {
        public bool CanBeAuthorized(string authenticationToken, string roleName)
        {
            var client = new RestClient("http://localhost:8080/identity");
            var request = new RestRequest("authentication/" + authenticationToken +
                                          "/inRole/" + roleName, Method.GET);
            var result = client.Execute(request);
            return result.StatusCode == HttpStatusCode.OK;
        }
    }
}

File: java/eshop-cqrs-ddd-microservices-excercise/common/src/main/java/com/marekturis/common/infrastructure/remote/RemoteAuthorizator.java
package com.marekturis.common.infrastructure.remote;

import com.marekturis.common.application.authorization.Authorizator;
import com.sun.jersey.api.client.Client;
import com.sun.jersey.api.client.ClientResponse;
import com.sun.jersey.api.client.WebResource;

import javax.inject.Named;

/**
 * @author Marek Turis
 */
@Named
public class RemoteAuthorizator implements Authorizator {

	@Override
	public boolean canBeAuthorized(String authenticationToken, String roleName) {
		Client client = Client.create();

		WebResource webResource = client.resource(ServiceLocations.IDENTITY +
				"authentication/" + authenticationToken +
				"/inRole/" + roleName);

		ClientResponse response = webResource.accept("application/json").get(ClientResponse.class);

		return response.getStatus() == 200;
	}
}
